src/ethereum/state_oracle/interface.py

```diff
@@ -0,0 +1,213 @@
+"""
+Abstract interface for state oracle implementations.
+"""
+
+from typing import Any, Callable, Optional, Protocol
+
+from ethereum_types.bytes import Bytes20, Bytes32
+from ethereum_types.numeric import U256
+
+# Use generic types for compatibility across forks
+Account = Any
+Address = Bytes20
+
+
+class MerkleOracle(Protocol):
+    """
+    Oracle interface for Merkle Patricia Trie based state.
+    """
+
+    def get_account(
+        self, address: Address  # noqa: U100
+    ) -> Account:  # noqa: U100
```

Comment:
I think if you `raise NotImplementedError` it mutes the linter, but I'm not 100% sure on that.